src/tools/crossgen2/ILCompiler.DependencyAnalysisFramework/PerfEventSource.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.Tracing;
+
+namespace ILCompiler.DependencyAnalysisFramework
+{
+    [EventSource(Name = "Microsoft-ILCompiler-Perf")]
+    public class PerfEventSource : EventSource
+    {
+        [Event(1, Level = EventLevel.Informational)]
+        public void LoadingStart() { WriteEvent(1); }
+        [Event(2, Level = EventLevel.Informational)]
+        public void LoadingStop() { WriteEvent(2); }
+
+        [Event(3, Level = EventLevel.Informational)]
+        public void GraphProcessingStart() { WriteEvent(3); }
+        [Event(4, Level = EventLevel.Informational)]
+        public void GraphProcessingStop() { WriteEvent(4); }
+
+        [Event(5, Level = EventLevel.Informational)]
+        public void EmittingStart() { WriteEvent(5); }
+        [Event(6, Level = EventLevel.Informational)]
+        public void EmittingStop() { WriteEvent(6); }
+
+        [Event(7, Level = EventLevel.Informational)]
+        public void CompilationStart() { WriteEvent(7); }
+        [Event(8, Level = EventLevel.Informational)]
+        public void CompilationStop() { WriteEvent(8); }
+
+        [Event(9, Level = EventLevel.Informational)]
+        public void JitStart() { WriteEvent(9); }
```

Comment:
ILCompiler.DependencyAnalysisFramework is a general purpose dependency graph analysis library (a separate component). Adding concepts like `Jit`, and `Emitting` is a bit of a hack - those concepts don't have a meaning in the general purpose library. I assume this file is going to collect a lot more of these things over time.

If we think there's value in instrumenting dependency analysis, we should just add another event source class for it. I think we can come up with other general purpose events that are worth tracking in ILCompiler.DependencyAnalyis (e.g. what types of nodes are we marking, how long it takes to collect dependencies for those nodes, etc.)

Reply:
This class handles more than just dependency analysis though. Would it be better to make a new ILCompiler.Instrumentation or ILCompiler.General project and put this there?

Reply:
It's about separation of concerns. ILCompiler.DependencyAnalysis doesn't know/care what ILCompiler.ReadyToRun does with it.

Moving a class that has the "Jit" or "Emitting" related events into a ILCompiler.General and having ILCompiler.DependencyAnalysis depend on it still violates the separation of concerns, but with extra steps.

We use ILCompiler.DependencyAnalysis elsewhere (not in this repo) and events specific to ILCompiler.ReadyToRun don't make sense in those places. Placing events that are relevant to _one of the consumers_ of ILCompiler.DependencyAnalysis into ILCompiler.DependencyAnalysis sets a bad precedent. Adding things that are convenient to dump stuff into tend to get misused (because they're convenient).

Is there any problems with introducing two separate EventSources to instrument the two separate components?

Reply:
Oh I see what you mean now. Sure I'll separate it.